client/dashboard/sites/settings-static-file-404/index.tsx

```diff
@@ -0,0 +1,155 @@
+import { DataForm } from '@automattic/dataviews';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Card,
+	CardBody,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { settings } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { siteQuery, siteStaticFile404Query, siteStaticFile404Mutation } from '../../app/queries';
+import { Callout } from '../../components/callout';
+import PageLayout from '../../components/page-layout';
+import SettingsPageHeader from '../settings-page-header';
+import calloutIllustrationUrl from './callout-illustration.svg';
+import type { Site } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+export function canSetStaticFile404Handling( site: Site ) {
+	return site.is_wpcom_atomic;
+}
+
+const fields: Field< { setting: string } >[] = [
+	{
+		id: 'setting',
+		label: __( 'Response' ),
+		Edit: 'select',
+		description: __(
+			'Assets are images, fonts, JavaScript, and CSS files that web browsers request as part of loading a web page. This setting controls how the web server handles requests for missing asset files.'
+		),
+		elements: [
+			{ value: 'default', label: __( 'Default' ) },
+			{ value: 'lightweight', label: __( 'Send a lightweight File-Not-Found page' ) },
+			{ value: 'wordpress', label: __( 'Delegate request to WordPress' ) },
+		],
+	},
+];
+
+const form = {
+	type: 'regular' as const,
+	fields: [ 'setting' ],
+};
+
+export default function SiteStaticFile404Settings( { siteSlug }: { siteSlug: string } ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
+	const { data: setting } = useQuery( {
+		...siteStaticFile404Query( siteSlug ),
+		enabled: site && canSetStaticFile404Handling( site ),
+	} );
+	const mutation = useMutation( siteStaticFile404Mutation( siteSlug ) );
+
+	const [ formData, setFormData ] = useState< { setting: string } >( {
+		setting: setting ?? '',
+	} );
+
+	if ( ! site ) {
+		return null;
+	}
+
+	const isDirty = formData.setting !== setting;
+	const { isPending } = mutation;
+
+	const handleUpgradePlan = () => {
+		const params = new URLSearchParams();
+		params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+		window.location.href = `/checkout/${ encodeURIComponent(
```

Comment:
This is probably good enough for now, but we need to think about the post-checkout flow and atomic transfer. Also, the redirect_to param doesn't seem to work when clicking the close button in the checkout page.

Reply:
DOTCOM-13307

Reply:
Nice, that worked 👍